src/util/bip32.rs

```diff
@@ -526,10 +568,52 @@ mod tests {
 
     use network::constants::Network::{self, Bitcoin};
 
+    use super::parse_derivation_path;
     use super::{ChildNumber, ExtendedPrivKey, ExtendedPubKey};
     use super::ChildNumber::{Hardened, Normal};
     use super::Error;
 
+    #[test]
+    fn test_parse_derivation_path() {
+        assert_eq!(parse_derivation_path("42"), Err(Error::InvalidDerivationPathFormat));
+        assert_eq!(parse_derivation_path("n/0'/0"), Err(Error::InvalidDerivationPathFormat));
+        assert_eq!(parse_derivation_path("m/0h/0x"), Err(Error::InvalidChildNumberFormat));
+
```

Comment:
suggestion: add test for something like `m//3/0'` with double-slashes as well has capital `H` for harden

Reply:
as well as something like `4/m/5` where `m` exists but isn't in first slot

Reply:
done, thanks!